addons/payment_mollie/models/payment_acquirer.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+import requests
+from werkzeug import urls
+
+from odoo import _, api, fields, models, service
+from odoo.exceptions import ValidationError
+
+from odoo.addons.payment_mollie.const import SUPPORTED_CURRENCIES
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentAcquirer(models.Model):
+    _inherit = 'payment.acquirer'
+
+    provider = fields.Selection(
+        selection_add=[('mollie', 'Mollie')], ondelete={'mollie': 'set default'}
+    )
+    mollie_api_key = fields.Char(
+        string="Mollie API Key",
+        help="The Test or Live API Key depending on the configuration of the acquirer",
+        required_if_provider="mollie", groups="base.group_system"
+    )
+
+    #=== BUSINESS METHODS ===#
+
+    @api.model
+    def _get_compatible_acquirers(self, *args, currency_id=None, **kwargs):
+        """ Override of payment to unlist Mollie acquirers for unsupported currencies. """
+        acquirers = super()._get_compatible_acquirers(*args, currency_id=currency_id, **kwargs)
+
+        currency = self.env['res.currency'].browse(currency_id).exists()
+        if currency and currency.name not in SUPPORTED_CURRENCIES:
+            acquirers = acquirers.filtered(lambda a: a.provider != 'mollie')
+
+        return acquirers
+
+    def _get_default_payment_method_id(self):
+        self.ensure_one()
+        if self.provider != 'mollie':
+            return super()._get_default_payment_method_id()
+        return self.env.ref('payment_mollie.payment_method_mollie').id
+
+    def _mollie_make_request(self, endpoint, data=None, method='POST'):
+        """ Make a request at mollie endpoint.
+
+        Note: self.ensure_one()
+
+        :param str endpoint: The endpoint to be reached by the request
+        :param dict data: The payload of the request
+        :param str method: The HTTP method of the request
+        :return The JSON-formatted content of the response
+        :rtype: dict
+        :raise: ValidationError if an HTTP error occurs
+        """
+        self.ensure_one()
+        endpoint = f'/v2/{endpoint.strip("/")}'
+        url = urls.url_join('https://api.mollie.com/', endpoint)
+
+        odoo_version = service.common.exp_version()['server_version']
```

Comment:
minor, but this is a complicated way of obtaining the `odoo.release.version` ;-)
```suggestion
        odoo_version = odoo.release.version
```
(after importing)